From 8137352dcc718b7bc874ddbcc29d56f9d3639864 Mon Sep 17 00:00:00 2001
From: Shuguang Gong <shuguang.gong@intel.com>
Date: Thu, 25 Jun 2015 15:26:47 +0800
Subject: [PATCH 436/462] ov2685: adjust mipi clk and enable 50Hz banding
 filter

1. enable 50Hz banding filter
2. decrease 2M mipi clk to limited range of ov2685

Enable 50Hz banding filter

Change-Id: If44403f08c741d1fc16d9946b00baac6f5f3fee9
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-40144
Signed-off-by: Shuguang Gong <shuguang.gong@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/61722
Reviewed-on: https://icggerrit.ir.intel.com/62186
Reviewed-by: Tian Shu Qiu <tian.shu.qiu@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Build: IMIN (SC)
Approver: Tian Shu Qiu <tian.shu.qiu@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../camera/drivers/media/i2c/ov2685.h              | 42 ++++++++++++++++++----
 1 file changed, 35 insertions(+), 7 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov2685.h b/drivers/external_drivers/camera/drivers/media/i2c/ov2685.h
index 7a253e3..7e2a28e 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov2685.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov2685.h
@@ -318,10 +318,10 @@ static struct ov2685_res_struct ov2685_res[] = {
 	.res	= OV2685_RES_2M,
 	.width	= 1600,
 	.height	= 1200,
-	.pixels_per_line = 1664,
-	.lines_per_frame = 1682,
+	.pixels_per_line = 1700,
+	.lines_per_frame = 1294,
 	.fps	= 30,
-	.pix_clk = 84,
+	.pix_clk = 66,
 	.used	= 0,
 	.regs	= NULL,
 	.skip_frames = 0,
@@ -479,7 +479,7 @@ static struct ov2685_reg const ov2685_2M_init[] = {
 	{OV2685_8BIT, 0x3018 , 0x84},
 	{OV2685_8BIT, 0x301d , 0xf0},
 	{OV2685_8BIT, 0x3020 , 0x00},
-	{OV2685_8BIT, 0x3082 , 0x46},/*mclk = 19.2Mhz*/
+	{OV2685_8BIT, 0x3082 , 0x37},/*mclk = 19.2Mhz*/
 	{OV2685_8BIT, 0x3083 , 0x03},
 	{OV2685_8BIT, 0x3084 , 0x07},
 	{OV2685_8BIT, 0x3085 , 0x03},
@@ -524,9 +524,9 @@ static struct ov2685_reg const ov2685_2M_init[] = {
 	{OV2685_8BIT, 0x380a , 0x04},
 	{OV2685_8BIT, 0x380b , 0xb0},
 	{OV2685_8BIT, 0x380c , 0x06},
-	{OV2685_8BIT, 0x380d , 0x80},
-	{OV2685_8BIT, 0x380e , 0x06},
-	{OV2685_8BIT, 0x380f , 0x92},
+	{OV2685_8BIT, 0x380d , 0xa4},
+	{OV2685_8BIT, 0x380e , 0x05},
+	{OV2685_8BIT, 0x380f , 0x0e},
 	{OV2685_8BIT, 0x3810 , 0x00},
 	{OV2685_8BIT, 0x3811 , 0x08},
 	{OV2685_8BIT, 0x3812 , 0x00},
@@ -536,7 +536,11 @@ static struct ov2685_reg const ov2685_2M_init[] = {
 	{OV2685_8BIT, 0x3819 , 0x04},
 	{OV2685_8BIT, 0x3820 , 0xc0},
 	{OV2685_8BIT, 0x3821 , 0x00},
+	{OV2685_8BIT, 0x382a , 0x08},
 	/* AIQ settings */
+	{OV2685_8BIT, 0x3a02 , 0x90},
+	{OV2685_8BIT, 0x3a03 , 0x4e},
+	{OV2685_8BIT, 0x3a04 , 0x40},
 	{OV2685_8BIT, 0x3a06 , 0x01},
 	{OV2685_8BIT, 0x3a07 , 0x84},
 	{OV2685_8BIT, 0x3a08 , 0x01},
@@ -788,7 +792,11 @@ static struct ov2685_reg const ov2685_1M3_init[] = {
 	{OV2685_8BIT, 0x3819 , 0x04},/* bin off */
 	{OV2685_8BIT, 0x3820 , 0xc0},
 	{OV2685_8BIT, 0x3821 , 0x00},
+	{OV2685_8BIT, 0x382a , 0x08},
 	/* AIQ settings */
+	{OV2685_8BIT, 0x3a02 , 0x90},
+	{OV2685_8BIT, 0x3a03 , 0x4e},
+	{OV2685_8BIT, 0x3a04 , 0x40},
 	{OV2685_8BIT, 0x3a06 , 0x01},
 	{OV2685_8BIT, 0x3a07 , 0x84},
 	{OV2685_8BIT, 0x3a08 , 0x01},
@@ -1039,7 +1047,11 @@ static struct ov2685_reg const ov2685_1408_1152_init[] = {
 	{OV2685_8BIT, 0x3815 , 0x11},
 	{OV2685_8BIT, 0x3819 , 0x04},
 	{OV2685_8BIT, 0x3820 , 0xc0},
+	{OV2685_8BIT, 0x382a , 0x08},
 	/* AIQ settings */
+	{OV2685_8BIT, 0x3a02 , 0x90},
+	{OV2685_8BIT, 0x3a03 , 0x4e},
+	{OV2685_8BIT, 0x3a04 , 0x40},
 	{OV2685_8BIT, 0x3821 , 0x00},
 	{OV2685_8BIT, 0x3a06 , 0x01},
 	{OV2685_8BIT, 0x3a07 , 0x84},
@@ -1295,7 +1307,11 @@ static struct ov2685_reg const ov2685_1468_1200_init[] = {
 	{OV2685_8BIT, 0x3819 , 0x04},
 	{OV2685_8BIT, 0x3820 , 0xc0},
 	{OV2685_8BIT, 0x3821 , 0x00},
+	{OV2685_8BIT, 0x382a , 0x08},
 	/* AIQ settings */
+	{OV2685_8BIT, 0x3a02 , 0x90},
+	{OV2685_8BIT, 0x3a03 , 0x4e},
+	{OV2685_8BIT, 0x3a04 , 0x40},
 	{OV2685_8BIT, 0x3a06 , 0x01},
 	{OV2685_8BIT, 0x3a07 , 0x84},
 	{OV2685_8BIT, 0x3a08 , 0x01},
@@ -1550,7 +1566,11 @@ static struct ov2685_reg const ov2685_732_600_init[] = {
 	{OV2685_8BIT, 0x3819 , 0x04},
 	{OV2685_8BIT, 0x3820 , 0xc2},
 	{OV2685_8BIT, 0x3821 , 0x01},
+	{OV2685_8BIT, 0x382a , 0x08},
 	/* AIQ settings */
+	{OV2685_8BIT, 0x3a02 , 0x90},
+	{OV2685_8BIT, 0x3a03 , 0x4e},
+	{OV2685_8BIT, 0x3a04 , 0x40},
 	{OV2685_8BIT, 0x3a06 , 0x01},
 	{OV2685_8BIT, 0x3a07 , 0x84},
 	{OV2685_8BIT, 0x3a08 , 0x01},
@@ -1802,7 +1822,11 @@ static struct ov2685_reg const ov2685_720p_init[] = {
 	{OV2685_8BIT, 0x3819 , 0x04},
 	{OV2685_8BIT, 0x3820 , 0xc0},
 	{OV2685_8BIT, 0x3821 , 0x00},
+	{OV2685_8BIT, 0x382a , 0x08},
 	/* AIQ settings */
+	{OV2685_8BIT, 0x3a02 , 0x90},
+	{OV2685_8BIT, 0x3a03 , 0x4e},
+	{OV2685_8BIT, 0x3a04 , 0x40},
 	{OV2685_8BIT, 0x3a06 , 0x00},
 	{OV2685_8BIT, 0x3a07 , 0xe4},
 	{OV2685_8BIT, 0x3a08 , 0x00},
@@ -2054,7 +2078,11 @@ static struct ov2685_reg const ov2685_svga_init[] = {
 	{OV2685_8BIT, 0x3819 , 0x04},
 	{OV2685_8BIT, 0x3820 , 0xc2},
 	{OV2685_8BIT, 0x3821 , 0x01},
+	{OV2685_8BIT, 0x382a , 0x08},
 	/* AIQ settings */
+	{OV2685_8BIT, 0x3a02 , 0x90},
+	{OV2685_8BIT, 0x3a03 , 0x4e},
+	{OV2685_8BIT, 0x3a04 , 0x40},
 	{OV2685_8BIT, 0x3a06 , 0x01},
 	{OV2685_8BIT, 0x3a07 , 0x84},
 	{OV2685_8BIT, 0x3a08 , 0x01},
-- 
1.9.1

